test/images/agnhost/webhook/main.go

```diff
@@ -86,77 +146,105 @@ func serve(w http.ResponseWriter, r *http.Request, admit admitFunc) {
 
 	klog.V(2).Info(fmt.Sprintf("handling request: %s", body))
 
-	// The AdmissionReview that was sent to the webhook
-	requestedAdmissionReview := v1beta1.AdmissionReview{}
-
-	// The AdmissionReview that will be returned
-	responseAdmissionReview := v1beta1.AdmissionReview{}
-
 	deserializer := codecs.UniversalDeserializer()
-	if _, _, err := deserializer.Decode(body, nil, &requestedAdmissionReview); err != nil {
-		klog.Error(err)
-		responseAdmissionReview.Response = toAdmissionResponse(err)
-	} else {
-		// pass to admitFunc
-		responseAdmissionReview.Response = admit(requestedAdmissionReview)
+	obj, gvk, err := deserializer.Decode(body, nil, nil)
+	if err != nil {
+		msg := fmt.Sprintf("Request could not be decoded: %v", err)
+		klog.Error(msg)
+		http.Error(w, msg, http.StatusBadRequest)
+		return
 	}
 
-	// Return the same UID
-	responseAdmissionReview.Response.UID = requestedAdmissionReview.Request.UID
+	if gvk.Kind != "AdmissionReview" {
+		msg := fmt.Sprintf("Unsupported kind: %s", gvk.Kind)
+		klog.Error(msg)
+		http.Error(w, msg, http.StatusBadRequest)
+		return
+	}
 
-	klog.V(2).Info(fmt.Sprintf("sending response: %v", responseAdmissionReview.Response))
+	var responseObj runtime.Object
+	switch gvk.Version {
+	case v1beta1.SchemeGroupVersion.Version:
```

Comment:
this doesn't verify group, just kind (on line 158) and version (here or on line 177). I actually expected the switch cases to compare to a full gvk for v1beta1 or v1 AdmissionReview

Reply:
Full gvk compare sounds simple. I'll go with that.